src/Language/PureScript/AST/Declarations.hs

```diff
@@ -510,6 +521,19 @@ declSourceAnn (TypeInstanceDeclaration sa _ _ _ _ _) = sa
 declSourceSpan :: Declaration -> SourceSpan
 declSourceSpan = fst . declSourceAnn
 
+declName :: Declaration -> Maybe Name
+declName (DataDeclaration _ _ n _ _) = Just (TyName n)
+declName (TypeSynonymDeclaration _ n _ _) = Just (TyName n)
+declName (ValueDeclaration _ n _ _ _) = Just (IdentName n)
+declName (ExternDeclaration _ n _) = Just (IdentName n)
+declName (ExternDataDeclaration _ n _) = Just (TyName n)
+declName (ExternKindDeclaration _ n) = Just (KiName n)
+declName (ValueFixityDeclaration _ _ _ n) = Just (ValOpName n)
+declName (TypeFixityDeclaration _ _ _ n) = Just (TyOpName n)
+declName (TypeClassDeclaration _ n _ _ _ _) = Just (TyClassName n)
+declName (TypeInstanceDeclaration _ n _ _ _ _) = Just (IdentName n)
+declName _ = Nothing
```

Comment:
Depending on how many more there are I'd suggest to write the cases out. Wildcards like this one have caused the most annoying bugs...